test/tests/rackhd20/test_rackhd20_api_ucs.py

```diff
@@ -0,0 +1,297 @@
+        self.assertEqual(api_data['status'], 200,
+                         'Incorrect HTTP return code, expected 200, got:' + str(api_data['status']))
+        for catalog_workflow in catalog_workflows:
+            self.wait_utility(str(catalog_workflow), 0, "Catalog")
+        for node in api_data['json']:
+            if node['id'] not in self.INITIAL_NODES:
+                api_data = fit_common.rackhdapi('/api/2.0/nodes/' + node['id'], action="delete")
+                logs.info_1("Deleting Node: {0}. Status was: {1}".format(node['id'], api_data['status']))
+                time.sleep(self.MAX_WAIT_ON_DELETE)
+
+        api_data = fit_common.rackhdapi('/api/2.0/nodes')
+        self.assertEqual(api_data['status'], 200,
+                         'Incorrect HTTP return code, expected 200, got:' + str(api_data['status']))
+        temp = {}
+        for node in api_data['json']:
+            temp[node['id']] = node['name']
+        logs.info_1("Found {0}  nodes remaining after restoring the nodes, should be {1}, Remaining nodes: {2}"
+                    .format(len(temp), len(self.INITIAL_NODES), temp))
+
+    def restore_obms_utility(self):
+        logs.info_1("Restoring OBMs")
+        api_data = fit_common.rackhdapi('/api/2.0/obms')
+        self.assertEqual(api_data['status'], 200,
+                         'Incorrect HTTP return code, expected 200, got:' + str(api_data['status']))
+
+        for obm in api_data['json']:
+            if obm['id'] not in self.INITIAL_OBMS:
+                api_data = fit_common.rackhdapi('/api/2.0/obms/' + obm['id'], action="delete")
+                logs.info_1("Deleting OBM: {0}. Status was: {1}".format(obm['id'], str(api_data['status'])))
+                time.sleep(self.MAX_WAIT_ON_DELETE)
+        api_data = fit_common.rackhdapi('/api/2.0/obms')
+        self.assertEqual(api_data['status'], 200,
+                         'Incorrect HTTP return code, expected 200, got:' + str(api_data['status']))
+        temp = {}
+        for obm in api_data['json']:
+            temp[obm['id']] = obm['service']
+        logs.info_1("Found {0} ucs obms remaining after restoring the obms, should be {1}. Remaining OBMs: {2}"
+                    .format(len(temp), len(self.INITIAL_OBMS), temp))
+
+    def ucs_url_factory(self, api, identifier=None):
+        """
+        returns a fully qualified UCS API
+        :param api:UCS API
+        :param identifier: identify the ucs element in the catalog API
+        :return:
+        """
+        ucs_service = self.RACKHD_IP + ":" + self.UCS_PORT + "/"
+        ucs_manager = self.UCS_IP
+        if identifier is None:
+            url = "http://" + ucs_service + api + "?host=" + ucs_manager + " &user=ucspe&password=ucspe"
+        else:
+            url = "http://" + ucs_service + api + "?host=" + ucs_manager + " &user=ucspe&password=ucspe"\
+                  "&identifier=" + identifier
+        return url
+
+    def test_api_20_ucs_log_in(self):
```

Comment:
Should add a test that if ucs.ip or ucs.port is not defined, the test script should bail out.
test_check_usc_params():
  if 'ucs_ip' not in fitcfg() or if 'ucs_ip' not in fitcfg():
     assert.fail.
Rest of tests, then can add a depends line.
@depends(after=test_check_ucs_params)
that way they get skipped, if the test is run against a stack without ucs.

Or you could sys.exit the script at assert.fail, tho that isn't very clean.